java/src/main/java/com/logicalclocks/hsfs/FeatureGroup.java

```diff
@@ -48,33 +57,13 @@
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class FeatureGroup extends FeatureGroupBase {
```

Comment:
add `@Overwrite` to all public methods and make sure they are established as abstract in the base class. Furthermore, there are multiple methods, that are not framework specific, move them to base file. The best way to identify if it is frameworks dependent is just to remove all imports related to spark and if you have issues then it stays in this class.